Fix move config away from operators

dennisdickmann-digital-power opened the pull request:
moved config parameters for HDXAdm0Operator.py to hdx_adm_dags.py instead of passing and calling config

cubonacci-demo committed:
Added initial pytest

cubonacci-demo committed:
chore: Merged new_structure back into initial-pytest branch

cubonacci-demo committed:
Merge branch 'new_structure' into initial-pytest

dennisdickmann-digital-power committed:
moved config parameters for HDXAdm0Operator.py to hdx_adm_dags.py

dennisdickmann-digital-power committed:
moved config parameters for HDXAdm0Operator.py to hdx_adm_dags.py

janvdvegt commented on dags/hdx_adm_dags.py:
```diff
@@ -18,8 +18,15 @@ def create_hdx_adm0_dag(country, schedule_interval, catchup, config, default_arg
-        country=country,
-        config=config,
+        source='cod',
+        input_filename="/opt/data/test/yem_adm_govyem_cso_ochayemen_20191002_GPKG.zip",
```
These will come from the config later as well, right?

dennisdickmann-digital-power replied:
Yes correct

janvdvegt commented on plugins/pipeline_plugin/operators/HDXAdm0Operator.py:
```diff
@@ -4,23 +4,23 @@
-                       config=config)
+def transform_adm0(source: str, input_filename, schema_filename, output_filename, iso3, raw_data_dir,
+                   geoboundaries_adm0_raw, schema_mapping, crs, *args, **kwargs):
+    hdx_adm0.transform(source=source, input_filename=input_filename, schema_filename=schema_filename,
```
It's cleaner for multiline method calls to have a named argument per line

janvdvegt commented on plugins/pipeline_plugin/operators/HDXAdm0Operator.py:
```diff
@@ -4,23 +4,23 @@
+    def __init__(self, source: str, input_filename, schema_filename, output_filename, iso3, raw_data_dir,
+                 geoboundaries_adm0_raw, schema_mapping, crs, *args, **kwargs) -> None:
+        super().__init__(python_callable=transform_adm0,
+                         op_kwargs={"source": source, "input_filename": input_filename,
```
Same here, have a key:value pair per line

dennisdickmann-digital-power committed:
fixed parameter typo and yaml indent issue

dennisdickmann-digital-power committed:
fixed parameter typo and yaml indent issue

cubonacci-demo committed:
Initial test suite done

dennisdickmann-digital-power committed:
added OSMRoadsTransformOperator

dennisdickmann-digital-power committed:
Merge remote-tracking branch 'origin/new_structure' into feat-osm-transform-operator

dennisdickmann-digital-power committed:
Merge remote-tracking branch 'origin/fix-move-config-away-from-operators' into fix-move-config-away-from-operators

# Conflicts:
#	dags/config/config.yaml

dennisdickmann-digital-power committed:
- refactored hdx_adm0.py and hdx_adm1.py into hdx_adm_transform.py
- moved configuration from HDXAdm1Operator.py to dag definition in hdx_adm_dags.py

dennisdickmann-digital-power committed:
- changed dag id's to better naming convention
- moved operator functions into their class as methods
- moved HDXExtractOperator.py configuration to the dag definition

dennisdickmann-digital-power committed:
- moved configuration for HDXRoadsTransformOperator.py, OSMExtractOperator.py and OSMRoadsTransformOperator.py to the dag definition
- refactored transform/osm_roads.py
- moved directory/filepath joining to config_parser.py

dennisdickmann-digital-power committed:
- added configuration to config.yaml
- added config methods to config_parser.py
- removed static filenames -> now called from config_parser.py
